ffi/src/lib.rs

```diff
@@ -388,14 +516,45 @@ fn create_client_internal(
             let redis_error = err.into();
             errors::error_message(&redis_error)
         })?;
+
+    let is_subscriber = request.pubsub_subscriptions.is_some() && pubsub_callback as usize != 0;
+    let (push_tx, mut push_rx) = tokio::sync::mpsc::unbounded_channel();
+    let tx = match is_subscriber {
+        true => Some(push_tx),
+        false => None,
+    };
+
     let client = runtime
-        .block_on(GlideClient::new(ConnectionRequest::from(request), None))
+        .block_on(GlideClient::new(ConnectionRequest::from(request), tx))
         .map_err(|err| err.to_string())?;
+
+    // Create the client adapter that will be returned and used as conn_ptr
     let core = Arc::new(CommandExecutionCore {
-        client,
-        client_type,
+        client: client.clone(),
+        client_type: client_type.clone(),
     });
-    Ok(ClientAdapter { runtime, core })
+    let client_adapter = Arc::new(ClientAdapter { runtime, core });
+    // Clone client_adapter before moving it into the async block
+    let client_adapter_ptr = Arc::into_raw(client_adapter).addr();
+    let client_adapter = unsafe { Arc::from_raw(client_adapter_ptr as *mut ClientAdapter) };
```

Comment:
As @jonathanl-bq advised, we can get the pointer without consuming the variable
```suggestion
    let client_adapter_ptr = Arc::as_ptr(&client_adapter).addr();
```